### PR TITLE
Add more test object types and methods to discogapic test API

### PR DESCRIPTION
**Pull request by andreamlin**
See simplecompute.v1.json for test API changes.
- Add a method (`compute.dummyObjects.delete`) with a null response type
- Add a float field and a double field to a message type
- Add, in one message type, two fields with different names but same reference schemas

Changes in simplecompute_gapic.yaml and baseline files were automatically generated from running the test cases.

There are some hacks in the generator code put in just to avoid NPEs, but those will be removed after handling these corner cases is done. (Fixes in https://github.com/googleapis/toolkit/pull/1892).

### CODE REVIEW

**Comment by andreamlin** [comment]
PTAL @garrettjonesgoogle

File: src/test/java/com/google/api/codegen/testdata/discogapic/java/java_simplecompute.v1.json.baseline [hunk 1]
```diff
@@ -2365,88 +2590,352 @@ public final class Error implements ApiMessage {
...
+
+  public Double getPrecisionFloatie() {
+    return precisionFloatie;
+  }
+
+  public Address getAddress() {
+    return primaryAddress;
+  }
+
+  public Address getAddress() {
```
Comment (andreamlin) [inline src/test/java/com/google/api/codegen/testdata/discogapic/java/java_simplecompute.v1.json.baseline]: This duplicate naming issue will be fixed in https://github.com/googleapis/toolkit/pull/1892.

**Comment by andreamlin** [comment]
PTAL
